src/ActivationContext.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {
+  createContext,
+  useState,
+  useEffect,
+  FunctionComponent,
+  useContext,
+} from "react"
+import { AppState, Platform } from "react-native"
+
+import { isBluetoothEnabled } from "./gaen/nativeModule"
+import { isLocationEnabled } from "./gaen/nativeModule"
+import { doesDeviceSupportLocationlessScanning } from "./gaen/nativeModule"
+
+export interface ActivationState {
+  isBluetoothOn: boolean
+  isLocationOn: boolean
+  isLocationNeeded: boolean
+}
+
+const initialState: ActivationState = {
+  isBluetoothOn: false,
+  isLocationOn: false,
+  isLocationNeeded: true,
+}
+
+export const ActivationContext = createContext<ActivationState>(initialState)
+
+const ActivationProvider: FunctionComponent = ({ children }) => {
```

Comment:
Could we have a name like `SystemServicesProvider` probably not a big deal but I thought activation could be brought into another context. Food for thought, I think is a perfectly used name but maybe worth considering a variant.

Reply:
I like that name a lot more, will update!